libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -402,8 +410,15 @@ void AP_ICEngine::update(void)
         if (!should_run) {
             state = ICE_OFF;
         } else if (now - starter_last_run_ms >= starter_delay*1000) {
-            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Starting engine");
-            state = ICE_STARTING;
+            // check if we should retry starting the engine
+            if (max_crank_retry <= 0 || crank_retry_ct < max_crank_retry) {
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Starting engine");
+                state = ICE_STARTING;
+                crank_retry_ct++;
+            } else {                    
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Engine max crank attempts reached");
```

Comment:
This probably warrant more than a "info" severity level. Probably critical or error. The definitions are here: https://mavlink.io/en/messages/common.html#MAV_SEVERITY

Reply:
Tell that to "Uncommanded engine stop" 😆 

I also think this should be critical or error, but I thought I'd get pushback given the severity of the engine stop message

Reply:
WARN